### ISSUE TITLE
Make the Clean Sheet theme compatible with the OSX version of Eclipse

### ISSUE THREAD

**Issue by pmadridb** [fappel/xiliary#56]

**Comment by fappel** [comment]
Sorry for responding so late on this issue. As I mentioned in http://www.codeaffine.com/2015/12/17/xmas-update-for-the-clean-sheet-eclipse-design/#comment-129722  the reason why clean sheet is only available for Windows 10 is the lack of resources (hardware, software, and human). However, I might be able to do some initial steps towards platform independence by resolving the following tasks (currently focused on Macs as I don't have any requests for Linux).

Notice that I assume the native scrollbars on Mac sufficient from an esthetical point of view, which is why I would exclude the FlatScrollBar overlay mechanism from the theme on OSX. Luckily, this makes the undertaking much cheaper as it avoids a significant amount of work to make the adapter mechanism work on Macs...

[ ] Introduce platform specific fragments for com.codeaffine.eclipse.swt and com.codeaffine.eclipse.swt.theme plugins. The first one is needed to swap a few widget related classes that cause inheritance problems due to different constructors of their super classes. The second is needed to provide the FlatScrollBar overlay extensions only on Windows.
[ ] Fix test related problems. We can take it for sure that there will be some SWT related unit tests that won't run on OSX. As a first step, I would go with additional ignore annotations like the one used for GTK platform as long as the problems concern the scrollable overlay mechanism. Fragments might also impose some challenges to the test infrastructure.
[ ] Provide a clean sheet CSS base file that gets extended with Windows and OSX specific files. The Windows file would contain the additional FlatScrollBar style settings and OSX deviating style settings for colors, sizes and the like.
[ ] Ensure that the build provides all platform dependent artifacts properly in the xiliary p2 repository.

I will take a look at the infrastructure tasks this week. Keep fingers crossed and hope that this will work out. Once it is in place, we could see how the theme works out on OSX.

**Comment by hstaudacher** [comment]
I just saw you have a travis based build running. What I know is that travis can also test on OS X. See https://docs.travis-ci.com/user/multi-os/ 

Maybe this will help.

**Comment by fappel** [comment]
Holger, thanks for the hint. I will check it out :-)

**Comment by fappel** [comment]
Unfortunately, Travis OS X supports only Java 7. So it's not an option at the time being :-(

**Comment by fappel** [comment]
The Mac OS X build infrastructure is now in place. I created a build chain that triggers an additional build run of a separate OS-X-build branch by a forced push at the end of the development or master build. If the latter succeeds it concludes by updating the gh-pages p2 repository. The Java 7 installation gets updated to Java 8 using homebrew at the beginning of the build execution. The first version of Clean Sheet on Mac OS X is available for download/installation on the CI build repository at http://fappel.github.io/xiliary/development/. But note this is not an official release. So, be careful installing it in a production environment. There seems to be an issue loading the Source Code Pro font. However, I was able to fix a related test failure on Mac, so this might be resolved. Anyway, Holger and I will have an in depth look at this problem next week.
